### ISSUE TITLE
MD030: Allow dynamic configuration for alignment in multi-digit ordered lists 

### ISSUE THREAD

**Issue by waldyrious** [DavidAnson/markdownlint#499]
### Problem

There's no way to configure [MD030](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md030) to enforce a consistent spacing after the marker of ordered lists, while still allowing me to have a list like this:

```md
1.  Lorem
2.  Ipsum
3.  Dolor
4.  Sit
5.  Amet
6.  Consectitur
7.  Adipisci
8.  Elit
9.  Foo
10. Bar
```

I could set 

```yml
MD030:
  ol_single: 2
```

which would cause an error in line 10, or

```yml
MD030:
  ol_single: 1
```

Which would accept line 10, but causes an error in all the other lines.

### Proposed solution

I'd like there to be an additional setting for `MD030`'s `ol_single` that would allow me to align the numbered items once the lists expand to double (or triple) digits.

Perhaps something like this:

```yml
MD030:
  ol_single: aligned
```

which would accept both

```
1. Lorem
2. Ipsum
3. Dolor
```

and

```md
8.  Elit
9.  Foo
10. Bar
```

**Comment by skwde** [comment]
Yes, this would be really nice to have.

I think it would also be important to specify at which level it should be aligned.

E.g.

```yml
MD030:
  align_all_at: 3
```

or making it dependent on indent as e.g. specified in `MD007/ul-indent`.
